hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcessSupervisor.java

```diff
@@ -84,9 +81,12 @@ public void run() {
     private void restartUnhealthy() {
         runningProcesses.stream()
                 .filter(consumerProcess -> !isHealthy(consumerProcess))
-                .forEach(consumerProcess -> taskQueue.offer(Signal.of(
-                        Signal.SignalType.RESTART_UNHEALTHY, consumerProcess.getSubscriptionName())
-                ));
+                .forEach(consumerProcess -> {
+                    Signal restartUnhealthy = Signal.of(RESTART_UNHEALTHY, consumerProcess.getSubscriptionName());
+                    logger.info("Lost contact with consumer {}, last seen {}ms ago. Creating signal {}.",
+                            consumerProcess, consumerProcess.lastSeen(), restartUnhealthy);
+                    taskQueue.offer(Signal.of(RESTART_UNHEALTHY, consumerProcess.getSubscriptionName()));
```

Comment:
You lost the id here. Please change line to `taskQueue.offer(restartUnhealthy)`.